### PR TITLE
[Tables] GA Preparation

### PR DESCRIPTION
**Pull request by seankane-msft**
- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

File: sdk/data/aztables/client.go [hunk 1]
```diff
@@ -113,9 +113,9 @@ func createTableResponseFromGen(g *generated.TableClientCreateResponse) CreateTa
 	return CreateTableResponse{}
 }
 
-// Create creates the table with the tableName specified when NewClient was called. If the service returns a non-successful
+// CreateTable creates the table with the tableName specified when NewClient was called. If the service returns a non-successful
 // HTTP status code, the function returns an *azcore.ResponseError type. Specify nil for options if you want to use the default options.
-func (t *Client) Create(ctx context.Context, options *CreateTableOptions) (CreateTableResponse, error) {
+func (t *Client) CreateTable(ctx context.Context, options *CreateTableOptions) (CreateTableResponse, error) {
```
Comment (jhendrixMSFT) [inline sdk/data/aztables/client.go:118]: Looks like this was left out of the list of breaking changes in the changelog.

File: sdk/data/aztables/client.go [hunk 2]
```diff
@@ -251,22 +203,45 @@ func (p *ListEntitiesPager) NextPage(ctx context.Context) (ListEntitiesPageRespo
 // Top: The maximum number of entities that will be returned per page of results.
 // Note: This value does not limit the total number of results if NextPage is called on the returned Pager until it returns false.
 //
-// List returns a Pager, which allows iteration through each page of results. Use nil for listOptions if you want to use the default options.
+// ListEntities returns a Pager, which allows iteration through each page of results. Use nil for listOptions if you want to use the default options.
 // For more information about writing query strings, check out:
 //  - API Documentation: https://docs.microsoft.com/en-us/rest/api/storageservices/querying-tables-and-entities
 //  - README samples: https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/data/aztables/README.md#writing-filters
-func (t *Client) List(listOptions *ListEntitiesOptions) ListEntitiesPager {
+func (t *Client) ListEntities(listOptions *ListEntitiesOptions) *runtime.Pager[ListEntitiesResponse] {
```
Comment (jhendrixMSFT) [inline sdk/data/aztables/client.go:210]: Also missing from the changelog?

File: sdk/data/aztables/constants.go [hunk 3]
```diff
@@ -17,40 +17,19 @@ const (
...
-}
-
-// ToPtr returns a *ResponseFormat pointing to the current value.
-func (c ResponseFormat) ToPtr() *ResponseFormat {
-	return &c
-}
-
-// EntityUpdateMode specifies what type of update to do on InsertEntity or UpdateEntity. ReplaceEntity
+// UpdateMode specifies what type of update to do on InsertEntity or UpdateEntity. ReplaceEntity
 // will replace an existing entity, MergeEntity will merge properties of the entities.
```
Comment (jhendrixMSFT) [inline sdk/data/aztables/constants.go:21]: Comment needs to be updated to reflect names.

File: sdk/data/aztables/entity.go [hunk 4]
```diff
@@ -150,6 +150,10 @@ func (e *EDMBinary) UnmarshalText(data []byte) error {
 	return nil
 }
 
+func (e *EDMBinary) ToByteSlice() []byte {
```
Comment (jhendrixMSFT) [inline sdk/data/aztables/entity.go:153]: Why was this added?  Same question applies to `ToInt64` and `ToString` below.

File: sdk/data/aztables/internal/constants.go [hunk 5]
```diff
@@ -10,7 +10,7 @@ package internal
 
 const (
 	moduleName    = "internal"
-	moduleVersion = "v0.6.1"
+	moduleVersion = "v0.5.1"
```
Comment (jhendrixMSFT) [inline sdk/data/aztables/internal/constants.go:13]: Version was decremented (unclear if this is actually used anywhere).

File: sdk/data/aztables/service_client.go [hunk 6]
```diff
@@ -328,17 +286,18 @@ func (g *GetPropertiesOptions) toGenerated() *generated.ServiceClientGetProperti
...
-	// The set of CORS rules.
-	Cors []*CorsRule `xml:"Cors>CorsRule"`
+	Properties ServiceProperties
+	// // The set of CORS rules.
+	// Cors []*CorsRule `xml:"Cors>CorsRule"`
 
-	// A summary of request statistics grouped by API in hourly aggregates for tables.
-	HourMetrics *Metrics `xml:"HourMetrics"`
+	// // A summary of request statistics grouped by API in hourly aggregates for tables.
+	// HourMetrics *Metrics `xml:"HourMetrics"`
```
Comment (jhendrixMSFT) [inline sdk/data/aztables/service_client.go:294]: Are these going to be removed?

File: sdk/data/aztables/internal/constants.go (hunk 5 above)
Comment (seankane-msft) [reply sdk/data/aztables/internal/constants.go]: Fixed

File: sdk/data/aztables/autorest.md [hunk 7]
```diff
@@ -11,7 +11,7 @@ clear-output-folder: false
 output-folder: internal
 tag: package-2019-02
 credential-scope: none
-use: "@autorest/go@4.0.0-preview.36"
+use: "C:/github.com/autorest.go"
 module-version: 0.5.1
```
Comment (jhendrixMSFT) [inline sdk/data/aztables/autorest.md:15]: Does this need to be updated now, or will that happen once the official code generator has been released along with `azcore`?

Comment (seankane-msft) [reply sdk/data/aztables/autorest.md:15]: Once the autorest release is tagged I will update this

**Review by jhendrixMSFT — APPROVED** [review]

File: sdk/data/aztables/go.mod [hunk 8]
```diff
@@ -1,10 +1,29 @@
 module github.com/Azure/azure-sdk-for-go/sdk/data/aztables
 
-go 1.16
+go 1.18
+
+replace github.com/Azure/azure-sdk-for-go/sdk/azcore => ../../azcore
```
Comment (richardpark-msft) [inline sdk/data/aztables/go.mod:5]: Are we going to push a tagged version of azcore?

File: sdk/data/aztables/autorest.md [hunk 9]
```diff
@@ -11,7 +11,7 @@ clear-output-folder: false
 output-folder: internal
 tag: package-2019-02
 credential-scope: none
-use: "@autorest/go@4.0.0-preview.36"
+use: "C:/github.com/autorest.go"
```
Comment (richardpark-msft) [inline sdk/data/aztables/autorest.md:14]: This is an odd path, is this a correct thing to do? (ie, shouldn't there be some version pining happening here?)

File: sdk/data/aztables/go.mod (hunk 8 above)
Comment (seankane-msft) [reply sdk/data/aztables/go.mod:5]: Yes, once core is tagged I'll update this.

File: sdk/data/aztables/autorest.md (hunk 9 above)
Comment (seankane-msft) [reply sdk/data/aztables/autorest.md:14]: Once the latest autorest version is pinned I will update this
